packages/react/src/__tests__/index.tsx

```diff
@@ -9,7 +9,15 @@ import { ProgressivelyProviderProps } from "../types";
 import { useFlags } from "../useFlags";
 
 const FlaggedComponent = () => {
-  const { flags } = useFlags();
+  const { flags, error, isLoading } = useFlags();
+
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+
+  if (error) {
+    return <div>Error</div>;
+  }
```

Comment:
Goal is to test loading an error states, but in the Provider, these state are "concurrent".
I mean you can possibly have the following : `{ isLoading: true, error: { message: "some error" } }`. If the `loadFlags` was firing an error for example :)

It's an issue because user will need to combine conditions in order to know the current state of the request. If we have an error, it means that loading is hover.

It can be a solution to use [Finite State Machine](https://thisrobot.life/#why-finite-state-machines). Can be achieved without library but by defining a finite number of states and their transitions.